### PR TITLE
feat: Allow configurable particle selection and reproducible seeds for Geant4

### PR DESCRIPTION
**Pull request by benjaminhuth**
This PR does two improvements to the geant4 examples integration:

- Make particle selection configurable for `addGeant4` as for `addFatras`
- Allow reproducible seeds for the geant4 algorithm (set the seed for each event from an `ActsExamples::RandomNumbers` object)

I did the change for the examples binaries not in the most beautiful way (so they do compile, but not changed the command line options or so), since they are deprecated anyways...

### CODE REVIEW

File: Examples/Algorithms/Geant4/include/ActsExamples/Geant4/Geant4Simulation.hpp [hunk 1]
```diff
@@ -60,8 +61,8 @@ class Geant4Simulation final : public BareAlgorithm {
     // Name of the output collection: material tracks
     std::string outputMaterialTracks = "";
 
-    // The Geant4 seed being used
-    std::size_t seed = 123;
+    /// Random number service.
+    std::shared_ptr<const RandomNumbers> randomNumbers;
```
Comment (asalzburger) [inline Examples/Algorithms/Geant4/include/ActsExamples/Geant4/Geant4Simulation.hpp:65]: Good move, like this it fits into our setup.

**Review by asalzburger — APPROVED** [review]
Very clear PR from my side, should make Geant4 reproducible.

**Review by andiwand — APPROVED** [review]
very nice!

**Comment by paulgessinger** [comment]
I'm noticing that the `linux_examples_test` is taking very long on this PR. Usually this completes under 20min. Could this be a problem with the change here?

**Comment by benjaminhuth** [comment]
> I'm noticing that the linux_examples_test is taking very long on this PR. Usually this completes under 20min. Could this be a problem with the change here @benjaminhuth?

I'll have a look at this after my vacation, but that will be in 2 weeks...

**Comment by andiwand** [comment]
@paulgessinger I just tried to reproduce this locally. for me this version and `main` have the same performance. should we try to rerun the CI? maybe it was a coincidence

**Comment by paulgessinger** [comment]
I'll try rerunning it, but I think it timed out in a few commits on this PR.

**Comment by benjaminhuth** [comment]
I think I resolved the problem, there was an uninitialized `ActsExamples::RandomNumbers` object which caused a segfault. Whats a bit strange is that a segfault seems to let the docker container freeze until the timeout...

Also, the root hashes of course change because of the changed seeds... 

I think that can be merged now!

**Review by andiwand — APPROVED** [review]

**Comment by paulgessinger** [comment]
@benjaminhuth can you update the hashes?

**Comment by benjaminhuth** [comment]
Could you re-approve @asalzburger so we can get this in?

**Comment by acts-project-service** [comment]
The backport to `develop/v19.x` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-develop/v19.x develop/v19.x
# Navigate to the new working tree
cd .worktrees/backport-develop/v19.x
# Create a new branch
git switch --create backport-1428-to-develop/v19.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick -x --mainline 1 9e22341ecb085e65471a0b4024c7d8f07477657f
# Push it to GitHub
git push --set-upstream origin backport-1428-to-develop/v19.x
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-develop/v19.x
```
Then, create a pull request where the `base` branch is `develop/v19.x` and the `compare`/`head` branch is `backport-1428-to-develop/v19.x`.

**Comment by acts-project-service** [comment]
The backport to `develop/v19.x` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-develop/v19.x develop/v19.x
# Navigate to the new working tree
cd .worktrees/backport-develop/v19.x
# Create a new branch
git switch --create backport-1428-to-develop/v19.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick -x --mainline 1 9e22341ecb085e65471a0b4024c7d8f07477657f
# Push it to GitHub
git push --set-upstream origin backport-1428-to-develop/v19.x
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-develop/v19.x
```
Then, create a pull request where the `base` branch is `develop/v19.x` and the `compare`/`head` branch is `backport-1428-to-develop/v19.x`.

**Comment by acts-project-service** [comment]
The backport to `develop/v19.x` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-develop/v19.x develop/v19.x
# Navigate to the new working tree
cd .worktrees/backport-develop/v19.x
# Create a new branch
git switch --create backport-1428-to-develop/v19.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick -x --mainline 1 9e22341ecb085e65471a0b4024c7d8f07477657f
# Push it to GitHub
git push --set-upstream origin backport-1428-to-develop/v19.x
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-develop/v19.x
```
Then, create a pull request where the `base` branch is `develop/v19.x` and the `compare`/`head` branch is `backport-1428-to-develop/v19.x`.

**Pull request merged**